pkgs/os-specific/linux/tuxedo-control-center/default.nix

```diff
@@ -0,0 +1,159 @@
+  });
+
+  nodePackages = baseNodePackages.package.override {
+    src = fetchFromGitHub {
+      owner = "tuxedocomputers";
+      repo = "tuxedo-control-center";
+      rev = "v${version}";
+      sha256 = "00gpsvjbli9f6vj31lbs7flmcz215r91iz6rg1kbzklc3y5k2ifj";
+    };
+
+    # Electron tries to download itself if this isn't set. We don't
+    # like that in nix so let's prevent it.
+    #
+    # This means we have to provide our own electron binaries when
+    # wrapping this program.
+    ELECTRON_SKIP_BINARY_DOWNLOAD=1;
+
+    # Angular prompts for analytics, which in turn fails the build.
+    #
+    # We can disable analytics using false or empty string
+    # (See https://github.com/angular/angular-cli/blob/1a39c5202a6fe159f8d7db85a1c186176240e437/packages/angular/cli/models/analytics.ts#L506)
+    NG_CLI_ANALYTICS="false";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "tuxedo-control-center";
+    exec = "tuxedo-control-center %U";
+    comment = "An application helping you to tune your TUXEDO";
+    desktopName = "TUXEDO Control Center";
+    icon = "tuxedo-control-center";
+    categories = "System;";
+  };
+in
+
+stdenv.mkDerivation rec {
+  name = "${baseName}-${version}";
+  src = "${nodePackages}/lib/node_modules/tuxedo-control-center/";
+
+  nativeBuildInputs = [
+    desktop-file-utils # for desktop-file-validate
+  ];
+
+  buildInputs = [
+    nodejs
+    makeWrapper
+
+    # For node-gyp
+    python
+  ];
+
+  # TCC by default writes its config to /etc/tcc, which is
+  # inconvenient. Change this to a more standard location.
+  #
+  # It also hardcodes binary path.
+  postPatch = ''
+    substituteInPlace src/common/classes/TccPaths.ts \
+      --replace "/etc/tcc" "/var/lib/tcc" \
+      --replace "/opt/tuxedo-control-center/resources/dist/tuxedo-control-center/data/service/tccd" "$out/bin/tccd"
+   '';
+
+  buildPhase = ''
+    set -x
+    # We already have `node_modules` in the current directory but we
+    # need it's binaries on `PATH` so we can use them!
+    export PATH="./node_modules/.bin:$PATH"
+
+    # Prevent npm from checking for updates
+    export NO_UPDATE_NOTIFIER=true
+
+    # The order of `npm` commands matches what `npm run build-prod` does but we split
+    # it out so we can customise the native builds in `npm run build-service`.
+    npm run clean
+    npm run build-electron
+
+    # We don't use `npm run build-service` here because it uses `pkg` which packages
+    # node binaries in a way unsuitable for nix. Instead we're doing it ourself.
+    tsc -p ./src/service-app
+    cp ./src/package.json ./dist/tuxedo-control-center/service-app/package.json
+
+    # We need to tell npm where to find node or `node-gyp` will try to download it.
+    # This also _needs_ to be lowercase or `npm` won't detect it
+    export npm_config_nodedir=${nodejs}
+    npm run build-native   # Builds to ./build/Release/TuxedoIOAPI.node
+
+    npm run build-ng-prod
+  '';
+
+  installPhase = ''
+    mkdir -p $out
```

Comment:
```suggestion
    runHook preInstall

    mkdir -p $out
```

Reply:
I've added pre/post hooks to build and install.